benchmarks/profiler/profile.go

```diff
@@ -1,24 +1,19 @@
-package main
+package profiler
 
 import (
 	"fmt"
-	_ "net/http/pprof"
 
+	"github.com/estensen/runtime-systems/benchmarks/fibonacci"
 	"github.com/pkg/profile"
 )
 
-func fibonacci(n int) {
-	a := 0
-	b := 1
-	for i := 0; i < n; i++ {
-		a, b = b, a+b
-
-		fmt.Println(b)
-	}
+//Profiler running from main
+func Profiler() {
+	fmt.Println("Running Fibonacci Profiler")
+	runFibonacciProfiler()
 }
 
-func main() {
+func runFibonacciProfiler() {
```

Comment:
This still means that we have to create a function for each program we're profiling. Can you make it so that the profile pkg takes a program in as an arg so we can generalize it?

Reply:
The reason why I'm calling Profile from main(), is that I thought you wanted Profile as its own package since you created a directory for it. The reason why I have Profiler, and than runFibonacci, was that I thought it would be easier to run the programs separately. 

Also, I thought the profiling starts when we write "defer profile.Start...", so if we start it right before calling a program it means that we're profiling the following program, or?

Reply:
Yes, I think it makes sense for `profile` to be a package. I think what you've done works as a step towards generalization. When we're gonna profile other packages we will probably retrieve them with `go get <pkg>`. So calling our profiler an arg: `go run profile.go <pkg>` and use the arg to call that pkg.

What if we changed `runFibonacciProfiler` to profilePackage`?

Reply:
The way `defer` works is that it defers the execution of a function until the surrounding function returns.